src/components/layout/NavBar.jsx

```diff
@@ -25,13 +30,31 @@ const NavBar = () => (
           </NavLink>
         </li>
         <li>
-          <NavLink className="nav-link  text-white" to="/register">
-            SignUp
-          </NavLink>
+          <li>
+            <Modal title="SignUp" action="SignUp">
+              <SignupForm signUp={signUp} signupData={signupData} />
```

Comment:
Since we are using react with redux, we could take the react data-driven approach to reduce this sort of coupling between our components, For example, our Navbar does not need the Login modal in any way.

Instead of nesting it here, we could render it somewhere it makes sense say in the` indexPage` Container, then when the user clicks on the login link, we dispatch an action to the store and the modal picks it up and renders

Reply:
I think you meant the signup modal not login modal